nautobot/core/tests/test_graphql.py

```diff
@@ -100,6 +100,17 @@ def test_execute_query(self):
         self.assertFalse(resp["data"].get("error"))
         self.assertEqual(len(resp["data"]["query"]), Location.objects.all().count())
 
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_execute_query_with_custom_field_type_json(self):
+        custom_field = CustomField.objects.create(type=CustomFieldTypeChoices.TYPE_JSON, label="custom_json_field")
+        custom_field.content_types.set([ContentType.objects.get_for_model(Location)])
+        custom_field_data = {"custom_json_field": {"name": "Custom Example", "is_customfield": True}}
+        self.locations[0]._custom_field_data = custom_field_data
+        self.locations[0].save()
+        query = "query ($name: [String!]) { locations(name:$name) {name, _custom_field_data, cf_custom_json_field} }"
```

Comment:
You're not specifying a `key` when creating the `CustomField` - I'm a bit surprised that's being permitted at all, maybe it's working only because you're not calling `clean`/`validated_save` on it?